lib/pure/collections/tableimpl.nim

```diff
@@ -101,29 +96,16 @@ template delImpl() {.dirty.} =
 
 template clearImpl() {.dirty.} =
   for i in 0 ..< t.dataLen:
-    when compiles(t.data[i].hcode): # CountTable records don't contain a hcode
+    when t.data is KeyValuePairSeq: # CountTable records don't contain a hcode
```

Comment:
The previous version was easier to read.

Reply:
No it wasn't.

Reply:
yes it was. And now you've introduced another regression, see https://github.com/nim-lang/Nim/pull/13687#issuecomment-602307527
Hard coding the types like you're doing is error prone. If another type gets an hcode, it'll introduce a bug.